### PR TITLE
Use add-extensions instead of finish-args

### PR DESCRIPTION
**Pull request by gasinvein**
Solves #183, needs testing.

### CODE REVIEW

File: com.valvesoftware.Steam.json [hunk 1]
```diff
@@ -47,6 +32,29 @@
...
+        },
+        "org.freedesktop.Platform.Compat.i386.Debug": {
+            "directory": "lib/debug/lib/i386-linux-gnu",
+            "version": "18.08",
+            "no-autodownload": true
+        },
+        "org.freedesktop.Platform.GL32": {
+            "directory": "lib/i386-linux-gnu/GL",
+            "version": "1.4",
+            "versions": "1.6;1.4",
```
Comment (TingPing) [inline com.valvesoftware.Steam.json]: I know this is unrelated to your change but in another commit this should be fixed to: `"versions": "18.08;1.4"`

**Comment by nanonyme** [comment]
Looks good, definitely improves readability

File: com.valvesoftware.Steam.json (hunk 1 above)
Comment (gasinvein) [reply com.valvesoftware.Steam.json]: Seems like Steam was just updated, I'll wait until someone updates the sha256 sum, before adding new changes to this branch.

**Comment by nanonyme** [comment]
bot, build

File: com.valvesoftware.Steam.json (hunk 1 above)
Comment (nanonyme) [reply com.valvesoftware.Steam.json]: The versions were set by @valentindavid

**Comment by nanonyme** [comment]
Did you by the way test this locally like http://docs.flatpak.org/en/latest/first-build.html

**Comment by gasinvein** [comment]
Yes, of course. I also checked app metadata this build produces, and it seems to be the same as with finish-args.

File: com.valvesoftware.Steam.json (hunk 1 above)
Comment (nanonyme) [reply com.valvesoftware.Steam.json]: The new release looks like udev changes only. Let's ignore for now

**Comment by nanonyme** [comment]
Okay, sounds like we get final confirmation only by pushing this through the build system. Do you still want to do additions?

**Comment by nanonyme** [comment]
https://flathub.org/builds/#/builders/2/builds/3986

**Comment by nanonyme** [comment]
I'm asking because title says wip still

File: com.valvesoftware.Steam.json (hunk 1 above)
Comment (nanonyme) [reply com.valvesoftware.Steam.json]: Added https://github.com/flathub/com.valvesoftware.Steam/issues/193 though

**Comment by gasinvein** [comment]
@nanonyme what about checksum of `steam_latest.tar.gz`? For me flatpak-builder says it has changed,

**Comment by nanonyme** [comment]
I can update if you like that done first, see #196

**Comment by gasinvein** [comment]
@nanonyme please merge it, if possible. Otherwise I can't build this PR locally without changing checksum in my branch.

**Comment by nanonyme** [comment]
Can you please rebase instead of merging master afterwards?

**Comment by gasinvein** [comment]
Sure, but why?

**Comment by nanonyme** [comment]
Less meaningless commits in history

**Comment by gasinvein** [comment]
Ok then.

**Comment by nanonyme** [comment]
If Git is on bad mood, just merge

File: com.valvesoftware.Steam.json (hunk 1 above)
Comment (gasinvein) [reply com.valvesoftware.Steam.json]: @TingPing Done.

**Comment by gasinvein** [comment]
@nanonyme done.

**Comment by nanonyme** [comment]
bot, build

**Comment by nanonyme** [comment]
https://flathub.org/builds/#/builders/2/builds/3991

**Comment by gasinvein** [comment]
Is there any way to view app metadata created by buildbot?

**Comment by nanonyme** [comment]
Not sure. Does https://flathub.org/builds/#/builders/26/builds/480 help? (it's the sub-build)

**Comment by nanonyme** [comment]
FB: Running 'flatpak build-finish --command=steam-wrapper --share=ipc --socket=x11 --socket=pulseaudio --share=network --talk-name=org.gnome.SettingsDaemon --talk-name=org.gnome.SettingsDaemon.MediaKeys --system-talk-name=org.freedesktop.NetworkManager --talk-name=org.kde.StatusNotifierWatcher --system-talk-name=com.feralinteractive.GameMode --system-talk-name=org.freedesktop.UPower --filesystem=xdg-music:ro --filesystem=xdg-pictures:ro --device=all --allow=multiarch --allow=devel --persist=. --env=LC_NUMERIC=C --env=ALSA_CONFIG_PATH= --env=MESA_GLSL_CACHE_DIR= --extension=org.freedesktop.Platform.Compat.i386=directory=lib/i386-linux-gnu --extension=org.freedesktop.Platform.Compat.i386=version=18.08 --extension=org.freedesktop.Platform.Compat.i386.Debug=directory=lib/debug/lib/i386-linux-gnu --extension=org.freedesktop.Platform.Compat.i386.Debug=no-autodownload=true --extension=org.freedesktop.Platform.Compat.i386.Debug=version=18.08 --extension=org.freedesktop.Platform.GL32=directory=lib/i386-linux-gnu/GL --extension=org.freedesktop.Platform.GL32=no-autodownload=true --extension=org.freedesktop.Platform.GL32=subdirectories=true --extension=org.freedesktop.Platform.GL32=add-ld-path=lib --extension=org.freedesktop.Platform.GL32=download-if=active-gl-driver --extension=org.freedesktop.Platform.GL32=enable-if=active-gl-driver '--extension=org.freedesktop.Platform.GL32=merge-dirs=vulkan/icd.d;glvnd/egl_vendor.d' --extension=org.freedesktop.Platform.GL32=version=1.4 '--extension=org.freedesktop.Platform.GL32=versions=18.08;1.4' /srv/buildbot/worker/build-x86_64-7/com.valvesoftware.Steam/.flatpak-builder/rofiles/rofiles-NKjAPP'

**Comment by gasinvein** [comment]
I guess now we should carefuly compare this with `finish-args` on master.

**Comment by nanonyme** [comment]
FB: Running 'flatpak build-finish --command=steam-wrapper --share=ipc --socket=x11 --socket=pulseaudio --share=network --talk-name=org.gnome.SettingsDaemon --talk-name=org.gnome.SettingsDaemon.MediaKeys --system-talk-name=org.freedesktop.NetworkManager --talk-name=org.kde.StatusNotifierWatcher --system-talk-name=com.feralinteractive.GameMode --system-talk-name=org.freedesktop.UPower --filesystem=xdg-music:ro --filesystem=xdg-pictures:ro --device=all --allow=multiarch --allow=devel --persist=. --extension=org.freedesktop.Platform.Compat.i386=directory=lib/i386-linux-gnu --extension=org.freedesktop.Platform.Compat.i386=version=18.08 --extension=org.freedesktop.Platform.Compat.i386.Debug=directory=lib/debug/lib/i386-linux-gnu --extension=org.freedesktop.Platform.Compat.i386.Debug=version=18.08 --extension=org.freedesktop.Platform.Compat.i386.Debug=no-autodownload=true --extension=org.freedesktop.Platform.GL32=directory=lib/i386-linux-gnu/GL --extension=org.freedesktop.Platform.GL32=version=1.4 '--extension=org.freedesktop.Platform.GL32=versions=1.6;1.4' --extension=org.freedesktop.Platform.GL32=subdirectories=true --extension=org.freedesktop.Platform.GL32=no-autodownload=true --extension=org.freedesktop.Platform.GL32=autodelete=false --extension=org.freedesktop.Platform.GL32=add-ld-path=lib '--extension=org.freedesktop.Platform.GL32=merge-dirs=vulkan/icd.d;glvnd/egl_vendor.d' --extension=org.freedesktop.Platform.GL32=download-if=active-gl-driver --extension=org.freedesktop.Platform.GL32=enable-if=active-gl-driver --env=LC_NUMERIC=C --env=ALSA_CONFIG_PATH= --env=MESA_GLSL_CACHE_DIR= /srv/buildbot/worker/build-x86_64-6/com.valvesoftware.Steam/.flatpak-builder/rofiles/rofiles-gT2Z18'

this is from the build that updated Steam

**Comment by gasinvein** [comment]
It is. Strange. Anyway, I think it's wouldn't cause problems, as any consequense is `org.freedesktop.Platform.GL32` being deleted when user deletes Steam.

**Comment by nanonyme** [comment]
I'll cherry-pick and make new PR with your commits if it turns out this was a false alarm but the related bug seemed fairly evil. (IOW there is no refcounting for autodelete and you just delete the data for all apps resulting on redownloads)

**Comment by gasinvein** [comment]
As far as I know, there are no other apps which use `org.freedesktop.Platform.GL32` anyway.

**Comment by nanonyme** [comment]
I went through all the other definitions, literally nothing on Flathub except Steam sets autodelete to false, only true. Looks like base on commits in flatpak that it goes as following: autodelete must be explicitly declared *except* for debug and locale which get autodelete on by default

**Pull request merged**